services/dynamic-common.js

```diff
@@ -1,22 +1,54 @@
+/**
+ * Common functions and utilities for tasks related to dynamic badges.
+ *
+ * @module
+ */
+
 import Joi from 'joi'
 import toArray from '../core/base-service/to-array.js'
 import validate from '../core/base-service/validate.js'
 import { InvalidResponse } from './index.js'
 
+/**
+ * Map of error codes and their corresponding error messages.
+ *
+ * @type {object}
+ */
 const errorMessages = {
   404: 'resource not found',
 }
 
+/**
+ * Joi schema for validating individual value.
+ * Checks if the individual value is of type string or number.
+ *
+ * @type {object}
+ */
 const individualValueSchema = Joi.alternatives()
   .try(Joi.string(), Joi.number())
   .required()
 
+/**
+ * Joi schema for validating compound value.
+ * Checks if the compound value is of type individualValueSchema, array of individualValueSchema or empty array.
+ *
+ * @type {object}
+ */
 const compoundValueSchema = Joi.alternatives().try(
   individualValueSchema,
   Joi.array().items(individualValueSchema).required(),
   Joi.array().length(0)
 )
 
+/**
+ * Transforms and validates the data against compoundValueSchema.
```

Comment:
I think a better description of this function for the docs would be something like


"Extract value from data object by key and validate value against compoundValueSchema."

or

"Look up value in data object by key and validate value against compoundValueSchema."

Reply:
This looks better. I'll update the docs. Thanks

Reply:
> Thanks for continuing to document these - its a big help.

No worries, hopefully, we can finish the entire documentation this year :)